segregation/util/util.py

```diff
@@ -8,6 +8,24 @@
 import numpy as np
 import math
 
+def _nan_handle(df, fillna = False):
+    """Check if dataframe has nan values.
+    Either raise an error or replace nan with 0.0.
+    """
+    if (str(type(df)) == '<class \'geopandas.geodataframe.GeoDataFrame\'>'):
+        values = df.loc[:, df.columns != df._geometry_column_name].values
+    else:
+        values = df.values
+        
+    if np.any(np.isnan(values)) and fillna == False:
```

Comment:
this checks whether there are any nan values in any of the rows, right? In cases where the whole row is nan, we could just drop it, and raise only when we have mismatch in the length of input nans

though this is cool too

Reply:
> this checks whether there are any nan values in any of the rows, right? 

Right.

> In cases where the whole row is nan, we could just drop it, 

I think this is a good idea.

> and raise only when we have mismatch in the length of input nans

I think we would need to check if the nan is in the same row, because the same length checking independently doesn't guarantee that it is in the same row.

Reply:
`df.dropna(how='all')` should take care of that?

Reply:
> `df.dropna(how='all')` should take care of that?

Yes, this will do. Should we go for that? What do you think about https://github.com/pysal/segregation/pull/130#issuecomment-522137184?

Reply:
i dont have a strong preference